cores/esp32/WMath.cpp

```diff
@@ -36,25 +36,17 @@ void randomSeed(unsigned long seed)
     }
 }
 
-long random(long howbig)
+long random( long howsmall, long howbig );
+long random( long howbig )
 {
-    uint32_t x = esp_random();
-    uint64_t m = uint64_t(x) * uint64_t(howbig);
-    uint32_t l = uint32_t(m);
-    if (l < howbig) {
-        uint32_t t = -howbig;
-        if (t >= howbig) {
-            t -= howbig;
-            if (t >= howbig)
-                t %= howbig;
-        }
-        while (l < t) {
-            x = esp_random();
-            m = uint64_t(x) * uint64_t(howbig);
-            l = uint32_t(m);
-        }
-    }
-    return m >> 32;
+  if ( howbig == 0 )
+  {
+    return 0 ;
+  }
+  if (howbig < 0) {
+    return (random(0, -howbig));
```

Comment:
@SuGlider Lets say I call `random(-10)` but get result of 0 to 10.
Shouldn't this be `return -(random(0, -howbig));` so we invert the value in return back on what it should be?

Reply:
No... For some reason, this is how it works in Arduino mainstream.
But in Arduino (other boards), when calling random(-5), thre results are positive and between 0 and 4.

Reply:
I tested that with other 3 non-ESP32 Arduino frameworks to make sure it was not a "bug".
There are also some discussion in Arduino Forum about it and all point to returning positive values for any rand(long value)
The only way to get negative values is using range with rand(val1, val2)

Reply:
Okey thanks for nice explanation 👍